test/unit/app/models/statistics_announcement_test.rb

```diff
@@ -292,6 +270,107 @@ class StatisticsAnnouncementTest < ActiveSupport::TestCase
     assert_equal statistics_announcement_date3.id, statistics_announcement.reload.current_release_date_id
   end
 
+  test "only valid when associated publication is of a matching type" do
+    statistics          = create(:draft_statistics)
+    national_statistics = create(:draft_national_statistics)
+    policy_paper        = create(:draft_policy_paper)
+
+    announcement = build(:statistics_announcement, publication_type_id: PublicationType::OfficialStatistics.id)
+
+    announcement.publication = statistics
+    assert announcement.valid?
+
+    announcement.publication = national_statistics
+    assert_not announcement.valid?
+    assert_equal ["type does not match announcement type: must be 'Official Statistics'"], announcement.errors[:publication]
+
+    announcement.publication_type_id = PublicationType::NationalStatistics.id
+    assert announcement.valid?
+
+    announcement.publication = policy_paper
+    assert_not announcement.valid?
+    assert_equal ["type does not match announcement type: must be 'Accredited Official Statistics'"], announcement.errors[:publication]
+  end
+
+  test "deleting statistics announcement does not delete publication" do
+    national_statistics = create(:draft_national_statistics, title: "Test")
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::NationalStatistics.id, publication: national_statistics)
+
+    announcement.delete
+    assert_equal Publication.find(national_statistics.id).title, "Test"
+    assert_equal StatisticsAnnouncement.where(id: announcement.id).count, 0
+  end
+
+  # === BEGIN: Publication Type update callback ===
+  test "should allow updating publication type with no attached publication" do
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::OfficialStatistics.id)
+
+    announcement.publication_type_id = PublicationType::NationalStatistics.id
+    announcement.save!
+
+    assert announcement.valid?
+    assert announcement.errors.empty?
+  end
+
+  test "should allow updating to same publication type with attached draft publication" do
```

Comment:
Feel like "same publication type *AS* attached draft publication" reads better. 
For this test, the starting type of the announcement is the same type we're changing to. Whilst the validation is still relevant, perhaps it makes more sense to start off with a non-accredited announcement for this test.

Reply:
hmmmm we can't have that starting state anyways, that's invalid as you cannot create an announcement type different to the publication type anyways. 

This test is more about updating the announcement, but the type has not been changed, so we're sending in the same publication type (as opposed to changing it)

Reply:
I thought this was the case that covers the "stuff got mixed up and now we're in the weird different type state" and we're trying to make the types the same, i.e. our zendesk ticket that started it all. 
Fair enough that we cannot do that on create, but then let's use update and replicate this scenario?

Reply:
but with your changes from the last PR, that scenario is just impossible. 

how to create mismatched models
- creation: cannot create with mismatched type because now validation is surfaced. also tested within this class
- update: that's what these suite of tests are testing, is that if you update it will automatically sync the types

so you will never get a starting state of mismatched types